app/javascript/components/GreetingContext.js

```diff
@@ -0,0 +1,23 @@
+import React, { createContext, useState, useEffect } from 'react';
+
+export const GreetingContext = createContext();
+
+export const GreetingProvider = ({ children }) => {
+  const [greeting, setGreeting] = useState('');
+
+  useEffect(() => {
+    async function fetchGreeting() {
+      const response = await fetch('/greetings/random');
+      const data = await response.json();
+      setGreeting(data.message);
+    }
+
+    fetchGreeting();
+  }, []);
+
```

Comment:
- [ ] Hi, You have done a great job in displaying random Greating, But I believe that you did not follow well the requirements. Learning objectives here is to **implement a connection between a Ruby on Rails back-end and a React front-end.** 
- The requirement says: **Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.**  I would advise you to follow this requirement.
- The picture below is the feedback I got when I changed the API in your react app to your localhost (a connection between a Ruby on Rails back-end and a React front-end)
![Capture](https://github.com/cancelei/hello-rails-react2/assets/63972670/598ec0b4-1fc8-4546-8537-c9a709df963f)